src/core/ext/census/resource.c

```diff
@@ -223,7 +223,9 @@ size_t allocate_resource(void) {
   if (n_resources == n_defined_resources) {
     size_t new_n_resources = n_resources ? n_resources * 2 : 2;
```

Comment:
while you are here, could you make the test over `n_resources` be `... n_resources != 0 ? ...`?